### DISCUSSION TITLE
Embedding Cutelyst into a Qt GUI Application

### DISCUSSION THREAD

**Discussion by bliepp** [cutelyst/cutelyst#339] [Q&A]
Hey guys,

I'm currently developing a GUI based software with the neccessity to provide some REST API + a landing page in a local network. As I'm using Qt anyway it seems convenient to use Cutelyst for this as the website states that it is embeddable into an existing application. Unfortunately I was not able to find **anything** in the Wiki about this and the API is also very confusing about this. Any ideas how to approach this problem?

Kindly
bliepp

**Comment by dantti** [comment]
You can embed a Cutelyst Application on any Qt app using the Server class. The Cutelyst::Server class is what Cutelystd process uses to provide it'a features.

**Reply by bliepp** [reply]
Any example on how to use it? There seems to be absolutely no documentation on this at all. The API does not even mention the Server class.
